driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java

```diff
@@ -0,0 +1,496 @@
+     * Besides, implementors are free to return {@code null} if access to the property through reflection is not required
+     * (in which case, they will likely have to provide a custom implementation of {@link MappedProperty}).
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                    either {@link Table @Table} or
+     *                    {@link UDT @UDT}.
+     * @param property    The property to locate a setter for; never {@code null}.
+     * @return The setter method for the given base class and given property, or {@code null} if no setter was found, or reflection is not required.
+     */
+    protected Method locateSetter(Class<?> mappedClass, PropertyDescriptor property) {
+        Method setter = property.getWriteMethod();
+        if (setter == null) {
+            // JAVA-984: look for a "relaxed" setter, ie. a setter whose return type may be anything
+            String propertyName = property.getName();
+            String setterName = "set" + propertyName.substring(0, 1).toUpperCase() + propertyName.substring(1);
+            try {
+                Method m = mappedClass.getMethod(setterName, property.getPropertyType());
+                if (!Modifier.isStatic(m.getModifiers()))
+                    setter = m;
+            } catch (NoSuchMethodException ignored) {
+            }
+        }
+        return setter;
+    }
+
+    /**
+     * Infers the Cassandra object name corresponding to given the property name.
+     * <p/>
+     * Most users should rely on the implementation provided here.
+     * It relies on annotation values and ultimately resorts to the
+     * {@link NamingStrategy} configured on this mapper.
+     * <p/>
+     * Subclasses may override this method if they need full control
+     * over generating Cassandra object names.
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                    either {@link Table @Table} or
+     *                    {@link UDT @UDT}.
+     * @param propertyName The property name; may not be {@code null} nor empty.
+     * @param annotations  The property annotations (found on its field and getter method); may not be {@code null} but can be empty.
+     * @return The inferred Cassandra object name.
+     */
+    protected String inferMappedName(@SuppressWarnings("unused") Class<?> mappedClass, String propertyName, Map<Class<? extends Annotation>, Annotation> annotations) {
+        if (annotations.containsKey(Computed.class)) {
+            String expression = ((Computed) annotations.get(Computed.class)).value();
+            if (expression.isEmpty())
+                throw new IllegalArgumentException(String.format("Property '%s': attribute 'value' of annotation @Computed is mandatory for computed properties", propertyName));
+            return expression;
+        }
+        boolean caseSensitive = false;
```

Comment:
In addition, as you mentioned `MapperCaseSensitivityTest.should_handle_case_sensitive_identifiers_when_keyspace_specified` is failing.  The reason is because The udt field name `zipCode` is being converted to `zipcode` because the default `NamingStrategy` enforces lowercasing of field name.

I think it makes sense that if a user doesn't provide a `NamingStrategy` explicitly instead we could have a default strategy that does nothing to the input fields when mapping to cassandra column (and vice versa).   We could then assume if a `NamingStrategy` *is* provided, we enforce case sensitivity, wdyt?